lib/ssh_scan/client.rb

```diff
@@ -21,13 +22,31 @@ def initialize(target, port)
     end
 
     def connect()
-      @sock = TCPSocket.new(@ip, @port)
-      @raw_server_banner = @sock.gets.chomp
-      @server_banner = SSHScan::Banner.read(@raw_server_banner)
-      @sock.puts(@client_banner.to_s)
+      begin
+        @sock = Socket.tcp(@ip, @port, connect_timeout: @timeout)
+      rescue Exception => e
```

Comment:
let's be more explicit about the exception type so we're not swallowing all exceptions from this call

Easiest way to test is to point it at something without a begin block around it and seeing what exception it will throw.